### PR TITLE
Scrub Paged Attention Serving Kernels and Obsolete Allocators

### PR DESCRIPTION
**Pull request by darisoy**
# Description

This PR completely removes and scrubs the obsolete, TPU-specific, and vulnerable custom JAX Pallas Paged Attention serving kernels and configurations from MaxText. During security reviews (such as `b/510375529`), the physical page manager allocator was flagged for potential cross-tenant HBM memory leaks. Following team alignment, it was confirmed that no production multimodal or Reinforcement Learning (RL) serving pipelines use `attention="paged"`.

To ensure model architecture definitions (Gemma, Llama, Mistral, Qwen, DeepSeek) and attention layer files do not crash on missing imports during this transition, this PR introduces lightweight, zero-overhead transitional compatibility shims in place of the deleted allocator and operator.

### Details & Implementation:
* **Paged Attention Deletion**: Purged custom Pallas kernels (`paged_attention_kernel_v2.py`) and page manager tests (`page_manager_test.py`). Scrubbed `pagedattn_` variables from `base.yml`, `types.py`, and config validators inside `pyconfig_deprecated.py`. Scrubbed page allocators and layout bindings inside `maxengine.py`.
* **Transitional Compatibility Shims**: Created dummy shims for `src/maxtext/inference/page_manager.py` (`PageState`, `PageManager`) and `src/maxtext/inference/paged_attention.py` (`PagedAttentionOp`) to allow all model layers to compile successfully with zero code mutations.

FIXES: b/510375529

# Tests

CI integration tests

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [X] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [X] I have necessary comments in my code, particularly in hard-to-understand areas.
- [X] I have run end-to-end tests tests and provided workload links above if applicable.
- [X] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

**Review by gobbleturk — COMMENTED** [review]
we should try deprecating paged attention instead of fixing it

File: src/maxtext/inference/maxengine/maxengine.py [hunk 1]
```diff
@@ -1337,7 +1419,10 @@ def _update_pages(prefix_page_idx, state):
     inserted_logits = jax.lax.with_sharding_constraint(inserted_logits, self.replicated_sharding)
     inserted_generated_tokens = jax.lax.with_sharding_constraint(inserted_generated_tokens, self.replicated_sharding)
     inserted_next_pos = jax.lax.with_sharding_constraint(inserted_next_pos, self.replicated_sharding)
-    inserted_tokens = jax.lax.with_sharding_constraint(inserted_tokens, self.replicated_sharding)
+    # JIT compatibility: preserve FSDP sharding
```
Comment (gobbleturk) [inline src/maxtext/inference/maxengine/maxengine.py]: we prefer not to have commented out code - can this be removed instead? I think we should aim to deprecate maxengine and rely on vLLM but I am not sure what the RL path looks for maxtext native models - @NicoGrande or @A9isha or @hengtaoguo may know if maxengine is used at all for any RL

**Review by gobbleturk — APPROVED** [review]

File: src/maxtext/inference/maxengine/maxengine.py (hunk 1 above)
Comment (NicoGrande) [reply src/maxtext/inference/maxengine/maxengine.py]: AFAIK maxengine is not used in RL, but still has some eval use cases (for models not yet supported in the vLLM codepath).

**Review by igorts-git — APPROVED** [review]

**Review by hengtaoguo — APPROVED** [review]

File: src/maxtext/inference/page_manager.py
Comment (jrplatin) [inline src/maxtext/inference/page_manager.py:1]: Why not just delete this file outright?

File: src/maxtext/inference/paged_attention.py
Comment (jrplatin) [inline src/maxtext/inference/paged_attention.py:1]: Same Q as above

**Review by jrplatin — APPROVED** [review]

File: src/maxtext/inference/page_manager.py
Comment (darisoy) [reply src/maxtext/inference/page_manager.py:1]: removed the file along with all of the references

File: src/maxtext/inference/paged_attention.py
Comment (darisoy) [reply src/maxtext/inference/paged_attention.py:1]: removed the file along with all of the references
